content/articles/a-look-at-c#-coding-conventions/index.md

```diff
@@ -0,0 +1,238 @@
+
+There are a few naming conventions that should be followed while writing C# code. All information about public elements applies as well to protected and protected internal elements in the following scenarios, which are all supposed to be visible to external callers.
+
+### Pascal's case
+In the Pascal case, every first letter of every word is capitalized as displayed in the example below. One can use the Pascal case for identifiers with three or more characters.
+
+When naming a class, record, or struct, use pascal casing ("PascalCasing").
+
+```
+PublicSchool1
+```
+
+```c#
+public class PascalCase1
+{
+}
+public record PublicIntitution1(
+    string Country,
+    string County,
+    string Village,
+    );
+public struct ValueCoordinate1
+{
+}
+```
+
+Besides to prefixing the name with an I, when naming an interface, use the pascal casing. Customers can tell it's a user interface because of this.
+
+```c#
+public interface IPublicInstitution
+{
+}
+```
+
+Besides to what it's already there, public members of types can be named using the pascal case.
+
+```c#
+public class PracticeExample1
+{
+    // public field
+    public bool IsValid;
+
+    // An init-only property is a property that allows you to do something only once.
+    public IBuilding Building { get; init; }
+
+    // An example of an event
+    public event Action EventProcessing;
+
+    // Implementation of a method
+    public void StartEventProcessing()
+    {
+        // Use of a local function
+        static int CountQueueItems() => WorkerQueue.Count;
+
+    }
+}
+```
+
+Since parameters are public properties of the record, a reader can use pascal casing when putting down positional accounts.
+
+```c#
+public record PublicIntitution
+(
+    string Country,
+    string County,
+    string Village,
+);
+```
+
+### Camel case
+The camelCasing convention, which is used for parameter names, capitalizes the first letter of each word except the first.
+
+```
+voidMain
+```
+
+Another implementation of the camel case is when naming private or internal fields.
+
+```
+camelCasing
+```
+
+```c#
+public class DataService
+{
+    private IWorkerQueue _workerQueue;
+}
+```
+
+It is also advisable to use the `s_prefix` and the `t_prefix` for the static fields respectively when dealing with thread and internal fields.
+
+```c#
+public class DataService
+{
+    private static IWorkerQueue s_workerQueue;
+
+    [ThreadStatic]
+    private static TimeSpan t_timeSpan;
+}
+```
+
+Camel casing is used when executing most of the method parameters.
+
+```c#
+public T SomeMethod<T>(int someNumber, bool isValid)
+{
+}
+```
+
+Apart from the above conventions, there are other several conventions that a programmer needs to consider and apply while in their daily routine of programming.
+1. The choice of word - Use readable identifier names. Underscores, hyphens, and other non-alphanumeric characters should not be used.
+2. Hungarian notation is neither acceptable nor encouraged.
```

Comment:
Explain further

Reply:
I have explained in the article above as follows
1. Choice of word- Not every word is used in programming but instead the correct identifiers and so are used together with the proper punctuation eg where and when to put a comma and such
2. Hungarian notation is neither acceptable nor encouraged - This is because a hungarian notation is a coding convention whose variable name depicts what it intends to do and if this is continually used it creates a programming language dependency and makes maintenance more difficult.